content/general/concepts/algorithm/terms/counting-sort/counting-sort.md

```diff
@@ -0,0 +1,121 @@
+---
+Title: 'Counting Sort'
+Description: 'Sorts a list with duplicate values efficiently.'
+Subjects:
+  - 'Computer Science'
+  - 'Interview Prep'
+Tags:
+  - 'Algorithms'
+  - 'Sorting Algorithms'
+  - 'Java'
+  - 'Counting Sort'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+**Counting sort** is an out-of-place, non-comparison sorting [algorithm](https://www.codecademy.com/resources/docs/general/algorithm) that sorts a list with duplicate values efficiently. It is a helpful algorithm for sorting [arrays](https://www.codecademy.com/resources/docs/java/arrays) of small, non-negative integers, where it analyzes elements' frequency distribution and places them in their final sorted positions.
+
+## Explanation
+
+1. Find the maximum value:
+
+- Iterate through the input array `data` to find the largest element `max`.
+  - This determines the range of values we need to count.
+
+2. Create the count array:
+
+- Initialize a new array `count` with a `max + 1` size to hold each value's frequency. 
+    - The size is `max + 1` so that we can include the `max` element.
+- Set all elements in `count` to _0_ initially.
+
+3. Count the occurrences:
+
+- Loop through the `data` array again.
+- For each element `data[i]`, increment the corresponding count in the `count` array (`count[data[i]]++`). 
+    - This is to document the frequency distribution of elements in `data`.
+
+4. Create the start array:
+
+- Initialize a new array `start` of the same size as `count`.
+- Set `start[0]` to _0_.
+- Based on the frequency distribution found in `count`, we can find the starting indexes and store them in the `start` array (`start[j] = start[j-1] + count[j-1]` for `j=1` to `max`).
```

Comment:
Avoid using first- and second-person pronouns (e.g. "I", "we", "you") if possible. Please refactor this.

Reply:
just removed the first person pronouns, thanks for the heads up